src/DotVVM.Framework/Configuration/UserColumnMappingCache.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using DotVVM.Framework.Controls;
+using DotVVM.Framework.ViewModel;
+
+namespace DotVVM.Framework.Configuration
+{
+    public class UserColumnMappingCache
+    {
+        private ConcurrentDictionary<Type, Dictionary<string, string>> mappingCache;
+        private IPropertySerialization propertySerialization;
```

Comment:
`IPropertySerialization` should not be used to figure out the property names. It can be renamed later using this API https://github.com/riganti/dotvvm/pull/262.

You should call `IViewModelSerializationMapper.GetMap` to get the real mapping of all properties.